backend/routes/reservation.js

```diff
@@ -137,4 +137,36 @@ router.delete('/single', async (req, res) => {
   res.json({ result: 'Deleted reservation' });
 });
 
+// This call gets the list of available tables for reservation
+router.get('/available', async (req, res) => {
+  const { date, time, numberOfGuests, restaurantID } = req.body;
+
+  if (!date || !time || !numberOfGuests || !restaurantID) {
+    res.status(400).json({
+      error:
+        'reservation/available GET endpoint needs: date, time, numberOfGuests and restaurantID body params'
+    });
+    return;
+  }
+
+  const { error, result } = await connection.asyncQuery(
+    'SELECT * ' +
+    'FROM restaurant_db.TABLE t ' +
+    'WHERE t.RestaurantID = ? AND t.maxGuests >= ? AND NOT EXISTS ( SELECT * ' +
```

Comment:
Are we ditching the minGuests? I am unsure what is for.

Reply:
It seems like a strange thing to check but I can add it

Reply:
Done